### PR TITLE
Updated API for MSL 0.11.1.0

### PR DESCRIPTION
**Pull request by Nylux**
Updated API documentation.

Added methods for:

- Events
- Table Injections
- Missing GML-related methods
- Mod Menu
- Disclaimer Rooms
- Loot Tables
- Rooms

### CODE REVIEW

File: docs/guides/api.en.md [hunk 1]
```diff
@@ -86,6 +89,63 @@
...
+    Msl.AddNewEvent("o_anotherobject", ModFiles.GetCode("myScript.gml"), EventType.Other, 24);
+
+    // Example 3 : Create Event from string, with object as variable
+    Msl.AddNewEvent(myVarContainingAGameObject, "scr_actionsLogUpdate(\"hello world\")", EventType.Create, 0);
+    ```
+    <h3>Arguments</h3>
+
+    | **Type** | **Name** | **Description** |
+    | :---: | :---: | :--- |
+    | `String` | `gameObject` | The name of the gameobject to add the event to. |
```
Comment (remyCases) [inline docs/guides/api.en.md:112]: how can we display the different signature (ie here we can either use a string or a UnderTaleGameObject as said in the previous example)

File: docs/guides/api.en.md [hunk 2]
```diff
@@ -86,6 +89,63 @@
...
+    | `Msl.EventType` | `eventType` | The type of event to add. |
+    | `uint` | `subtype` | The subtype of the event to add. </br>(Some events have different sub-events, like `Other` or `Draw`) |
+
+    <h3>Returns</h3>
+
+    | **Type** | **Description** |
+    | :--- | :--- |
+    | `N/A` | N/A |
+
+??? example "EventName `method`"
```
Comment (remyCases) [inline docs/guides/api.en.md:123]: Not sure if it's useful in the doc

File: docs/guides/api.en.md [hunk 3]
```diff
@@ -212,6 +297,26 @@
     | :--- | :--- |
     | `String` | The content of the code if found. |
 
+??? info "SetStringGMLInFile `method`"
```
Comment (remyCases) [inline docs/guides/api.en.md:300]: Maybe flag this as legacy ?

File: docs/guides/api.en.md [hunk 4]
```diff
@@ -345,7 +471,344 @@
...
-## Patching : Stage 1 (Loading)
+## Settings
+
+??? example "AddMenu `method`"
+    <h3>Summary</h3>
+    Adds an entry to the in-game mod menu, which lets users configure settings at runtime.
+    <h3>Example</h3>
+    ```c#
+    // Example : Adds a slider to the mod menu to configure the amount of tiles walked by turn.
+    // The slider will have a range from 2 to 5, with a default value of 4.
```
Comment (remyCases) [inline docs/guides/api.en.md:482]: Not sure if the example is clear or not. It does not configure the amount of tiles walked by turn but change the value holds by global.tiles_by_turn and the text displaid is "Tiles walked by turn".

Maybe using a non specific example would be better

File: docs/guides/api.en.md [hunk 5]
```diff
@@ -345,7 +471,344 @@
...
+
+## Loot
+
+??? example "AddLootTable `method`"
+    <h3>Summary</h3>
+    Creates a new loot table, which is a weighted list of gameobjects that can be found/dropped.
+    <h3>Example</h3>
+    ```c#
+    // Creates a new loot table with guaranteed items and items that have various chances of dropping.
+    // Also handles empty drops and items with rarity / durability.
```
Comment (remyCases) [inline docs/guides/api.en.md:548]: for later add what durability and rarity are and what can of values is expected in listItems

File: docs/guides/api.en.md [hunk 6]
```diff
@@ -345,7 +471,344 @@
...
+    <h3>Returns</h3>
+
+    | **Type** | **Description** |
+    | :--- | :--- |
+    | `N/A` | N/A |
+
+??? example "AddReferenceTable `method`"
+    <h3>Summary</h3>
+    Assigns a gameobject to a specific loot table.</br>
+    This means that any and all instances of the gameobject will drop items from the specified loot table.
```
Comment (remyCases) [inline docs/guides/api.en.md:587]: all instances (unless specified)

File: docs/guides/api.en.md [hunk 7]
```diff
@@ -345,7 +471,344 @@
...
+    ```
+    <h3>Returns</h3>
+
+    | **Type** | **Description** |
+    | :--- | :--- |
+    | `IEnumerable<UndertaleRoom>` | An iterable of all the rooms in the game. |
+
+??? example "AddRoom `method`"
+    <h3>Summary</h3>
+    Adds a room to the game.
```
Comment (remyCases) [inline docs/guides/api.en.md:698]: maybe we should explain what happens if already existing

File: docs/guides/api.en.md [hunk 8]
```diff
@@ -345,7 +471,344 @@
...
+
+    <h3>Returns</h3>
+
+    | **Type** | **Description** |
+    | :--- | :--- |
+    | `UndertaleLayer` | The layer found. |
+
+??? example "AddLayer `method`"
+    <h3>Summary</h3>
+    Adds a layer to a room.
```
Comment (remyCases) [inline docs/guides/api.en.md:746]: same as AddRoom

File: docs/guides/api.en.md [hunk 9]
```diff
@@ -345,7 +471,344 @@
...
+
+    <h3>Returns</h3>
+
+    | **Type** | **Description** |
+    | :--- | :--- |
+    | `UndertaleLayer` | The layer created. |
+
+??? example "AddGameObject `method`"
+    <h3>Summary</h3>
+    Adds a gameobject to a room.</br>
```
Comment (remyCases) [inline docs/guides/api.en.md:769]: same

**Review by remyCases — COMMENTED** [review]
Some clarifications to be added

File: docs/guides/api.en.md (hunk 1 above)
Comment (Nylux) [reply docs/guides/api.en.md:112]: Check the `Summary` section of the `Patching : Stage 2 (Matching)`, that's how I displayed it before

File: docs/guides/api.en.md (hunk 2 above)
Comment (Nylux) [reply docs/guides/api.en.md:123]: Fair, we can remove it if needed.

File: docs/guides/api.en.md (hunk 3 above)
Comment (Nylux) [reply docs/guides/api.en.md:300]: Yep, we do want people to use the chained methods.

File: docs/guides/api.en.md (hunk 4 above)
Comment (Nylux) [reply docs/guides/api.en.md:482]: More of an inaccuracy on what it does, but I agree, it would be better.

File: docs/guides/api.en.md (hunk 5 above)
Comment (Nylux) [reply docs/guides/api.en.md:548]: Leaving it up to you, you know best for that method.

File: docs/guides/api.en.md (hunk 7 above)
Comment (Nylux) [reply docs/guides/api.en.md:698]: Definitely.

File: docs/guides/api.en.md (hunk 2 above)
Comment (remyCases) [reply docs/guides/api.en.md:123]: done

**Pull request merged**